connect-location/src/main/java/com/ifttt/location/ConnectLocation.java

```diff
@@ -96,8 +100,75 @@ public static ConnectLocation getInstance() {
      * connection is enabled and location permissions need to be granted.
      */
     public void setUpWithConnectButton(ConnectButton connectButton, LocationPermissionCallback permissionCallback) {
-        this.permissionCallback = permissionCallback;
-        connectButton.addButtonStateChangeListener(this);
+        checkInit(this);
+
+        connectButton.addButtonStateChangeListener(new LocationButtonStateChangeListener(connectButton.getContext(),
+            permissionCallback
+        ));
+    }
+
+    /**
+     * Given the connection id passed in during initialization, fetch the connection data, and check if it has an
+     * enabled {@link UserFeature} that uses location.
+     *
+     * This method should be used in addition to the {@link #init(Context, String, ConnectionApiClient)} or
+     * {@link #init(Context, String, UserTokenProvider)} method in the initialization process, in order to set up
+     * ConnectLocation to account for users who have the connection already enabled but hasn't had it set up in your
+     * app. This method fetches the latest connection data, checks if, for the given user (via {@link UserTokenProvider},
+     * there is an enabled {@link UserFeature} that uses Location trigger. If that is true, it will set up the geofences
+     * and call the {@link LocationPermissionCallback#onRequestLocationPermission()}.
+     *
+     * If you have a connection that uses Location service, you should call this method in the first Activity that your
+     * users use your app, so that you can prompt the location permission request as soon as possible.
+     *
+     * @param activity Activity instance of the place where the permission check happens.
+     * @param permissionCallback Nullable {@link LocationPermissionCallback} instance, if non-null, it will be invoked
+     * when the connection is enabled, has an enabled UserFeature that has at least one Location service trigger, and
+     * the app doesn't have ACCESS_FINE_LOCATION permission.
+     *
+     * @return a {@link PendingResult} instance if we need to fetch the connection data, or null if we can reuse the
+     * cached data.
+     */
+    @Nullable
+    public PendingResult<Connection> checkLocationPermission(
+        Activity activity, @Nullable LocationPermissionCallback permissionCallback
+    ) {
+        checkInit(this);
```

Comment:
same comment as above regarding not having to perform this check here. This method is also invoked on the instance fetched by calling `getInstance()` first.